RecoLocalTracker/SiPixelRecHits/src/PixelCPEFast.cc

```diff
@@ -0,0 +1,594 @@
+  int ymin = theClusterParam.theCluster->minPixelCol();
+  int ymax = theClusterParam.theCluster->maxPixelCol();
+
+  // Iterate over the pixels.
+  int isize = theClusterParam.theCluster->size();
+  for (int i = 0; i != isize; ++i) {
+    auto const& pixel = theClusterParam.theCluster->pixel(i);
+    // ggiurgiu@fnal.gov: add pixel charge truncation
+    int pix_adc = pixel.adc;
+    if (truncate)
+      pix_adc = std::min(pix_adc, theClusterParam.pixmx);
+
+    //
+    // X projection
+    if (pixel.x == xmin)
+      Q_f_X += pix_adc;
+    if (pixel.x == xmax)
+      Q_l_X += pix_adc;
+    //
+    // Y projection
+    if (pixel.y == ymin)
+      Q_f_Y += pix_adc;
+    if (pixel.y == ymax)
+      Q_l_Y += pix_adc;
+  }
+}
+
+//==============  INFLATED ERROR AND ERRORS FROM DB BELOW  ================
+
+//-------------------------------------------------------------------------
+//  Hit error in the local frame
+//-------------------------------------------------------------------------
+LocalError PixelCPEFast::localError(DetParam const& theDetParam, ClusterParam& theClusterParamBase) const {
+  ClusterParamGeneric& theClusterParam = static_cast<ClusterParamGeneric&>(theClusterParamBase);
+
+  // Default errors are the maximum error used for edge clusters.
+  // These are determined by looking at residuals for edge clusters
+  float xerr = EdgeClusterErrorX_ * micronsToCm;
+  float yerr = EdgeClusterErrorY_ * micronsToCm;
+
+  // Find if cluster is at the module edge.
+  int maxPixelCol = theClusterParam.theCluster->maxPixelCol();
+  int maxPixelRow = theClusterParam.theCluster->maxPixelRow();
+  int minPixelCol = theClusterParam.theCluster->minPixelCol();
+  int minPixelRow = theClusterParam.theCluster->minPixelRow();
+
+  bool edgex = phase1PixelTopology::isEdgeX(minPixelRow) | phase1PixelTopology::isEdgeX(maxPixelRow);
+  bool edgey = phase1PixelTopology::isEdgeY(minPixelCol) | phase1PixelTopology::isEdgeY(maxPixelCol);
+
+  unsigned int sizex = theClusterParam.theCluster->sizeX();
+  unsigned int sizey = theClusterParam.theCluster->sizeY();
+
+  // Find if cluster contains double (big) pixels.
+  bool bigInX = theDetParam.theRecTopol->containsBigPixelInX(minPixelRow, maxPixelRow);
+  bool bigInY = theDetParam.theRecTopol->containsBigPixelInY(minPixelCol, maxPixelCol);
+
+  if (UseErrorsFromTemplates_) {
+    //
+    // Use template errors
+
+    if (!edgex) {  // Only use this for non-edge clusters
+      if (sizex == 1) {
+        if (!bigInX) {
+          xerr = theClusterParam.sx1;
+        } else {
+          xerr = theClusterParam.sx2;
+        }
+      } else {
+        xerr = theClusterParam.sigmax;
+      }
+    }
+
+    if (!edgey) {  // Only use for non-edge clusters
+      if (sizey == 1) {
+        if (!bigInY) {
+          yerr = theClusterParam.sy1;
+        } else {
+          yerr = theClusterParam.sy2;
+        }
+      } else {
+        yerr = theClusterParam.sigmay;
+      }
+    }
+
+  } else {  // simple errors
```

Comment:
the part here looks copy-pasted from PixelCPEGeneric.
Is it used? (the only variants are with `UseErrorsFromTemplates`)
something to factor out to reuse among the two CPEs or remove here if not needed

Reply:
Added to the post-merge open issues at https://github.com/cms-sw/cmssw/issues/32483 .